test/core/test_task_launcher.py

```diff
@@ -52,10 +65,12 @@ def test_init_on_task_run(self):
         self.assertEqual(len(launcher.units), 0)
         self.assertEqual(launcher.provider_type, MockProvider.PROVIDER_TYPE)
 
-    def test_create_launch_expire_assignments(self):
+    def test_create_expire_assignments(self):
         """Initialize a launcher on a task run, then create the assignments"""
         mock_data_array = self.get_mock_assignment_data_array()
-        launcher = TaskLauncher(self.db, self.task_run, mock_data_array)
+        launcher = TaskLauncher(
+            self.db, self.task_run, mock_data_array, max_num_concurrent_units=1
+        )
```

Comment:
You shouldn't be editing existing tests, but rather adding new ones to test the new functionality.

Reply:
yeah the additional parameter was meant for a test case in an existing one. But there were few lines in this test which fails the new way of units' statuses (it tested whether all of the units had `LAUNCHED` status after calling `launcher.launch_units()`